packages/common/src/types/projects.ts

```diff
@@ -252,6 +252,34 @@ export const maybeOverrideWarehouseConnection = <
     };
 };
 
+/**
+ * Merges new warehouse credentials with base credentials, preserving advanced settings
+ * like requireUserCredentials from the base credentials.
+ *
+ * This is useful when creating preview projects where we want to use new connection details
+ * (like from dbt profiles) but preserve advanced configuration from the parent project.
+ */
+export const mergeWarehouseCredentials = <T extends CreateWarehouseCredentials>(
+    baseCredentials: T,
+    newCredentials: T,
+): T => {
+    // If types don't match, return newCredentials as-is (can't merge different warehouse types)
+    if (baseCredentials.type !== newCredentials.type) {
+        return newCredentials;
+    }
+
+    // Merge credentials, with newCredentials taking precedence for connection details
+    // but baseCredentials providing advanced settings like requireUserCredentials
+    const merged = {
+        ...baseCredentials,
+        ...newCredentials,
+        // Preserve requireUserCredentials from base if not explicitly set in new credentials
```

Comment:
I think the comment just needs to be updated, because it communicates the opposite of what's happening. Or even just remove the comment.